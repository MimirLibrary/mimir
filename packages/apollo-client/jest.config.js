module.exports = {
  displayName: 'apollo-client',
  preset: '../../jest.preset.js',
  coverageDirectory: '../../coverage/packages/apollo-client',
};
