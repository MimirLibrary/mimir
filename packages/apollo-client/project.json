{
  "sourceRoot": "packages/apollo-client/src",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/apollo-client",
        "main": "packages/apollo-client/src/index.ts",
        "tsConfig": "packages/apollo-client/tsconfig.lib.json",
        "assets": ["packages/apollo-client/*.md"]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/apollo-client/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/packages/apollo-client"],
      "options": {
        "jestConfig": "packages/apollo-client/jest.config.js",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
