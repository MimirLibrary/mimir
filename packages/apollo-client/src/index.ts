export * from './lib/instance';
export * from './lib/generated/client-types';
