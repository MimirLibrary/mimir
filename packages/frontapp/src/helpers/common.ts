export const shortenText = (text: string, numberToShorten: number) => {
  return text.trim().slice(0, numberToShorten);
};
