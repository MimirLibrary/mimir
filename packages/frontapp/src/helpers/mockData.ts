export const mockStatus = [
  {
    person_id: 1,
    status: 'Free',
    created_at: '2022-09-08T03:58:39.228Z',
    id: 1,
    material_id: 1,
  },
];

export const mockMaterial = [
  {
    __typename: 'Material',
    author: 'test Author',
    category: 'test Category',
    created_at: '2022-09-08T03:58:39.228Z',
    description: 'Test Description',
    id: '1',
    id_type: 'Test Id_Type',
    identifier: '1111111111',
    is_donated: false,
    notification: [],
    picture: '',
    statuses: mockStatus,
    title: 'test',
    type: 'Book',
    updated_at: '2022-09-08T03:58:39.228Z',
  },
];
