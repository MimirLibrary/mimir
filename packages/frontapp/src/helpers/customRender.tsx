import '@testing-library/jest-dom';
import { render, RenderOptions } from '@testing-library/react';
import React, { FC, ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../app/store';
import { ApolloProvider } from '@apollo/client';
import { client } from '@mimir/apollo-client';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string | null) => str,
      i18n: {
        changeLanguage: () =>
          new Promise(() => {
            return;
          }),
      },
    };
  },
}));

jest.unmock('react-redux');

const wrapper: FC<{ children: React.ReactNode }> = ({ children }) => {
  return (
    <BrowserRouter>
      <ApolloProvider client={client}>
        <Provider store={store}>{children}</Provider>
      </ApolloProvider>
    </BrowserRouter>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => {
  return render(ui, { wrapper, ...options });
};

export { customRender as render };
