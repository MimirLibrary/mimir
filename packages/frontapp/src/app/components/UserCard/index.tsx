import BackButton from '../BackButton';
import { useParams } from 'react-router-dom';
import styled from '@emotion/styled';
import { colors, dimensions } from '@mimir/ui-kit';
import {
  GetOnePersonDocument,
  useChangePersonRoleMutation,
  useCreateStateMutation,
  useGetOnePersonQuery,
} from '@mimir/apollo-client';
import { mockData } from '../UserList/mockData';
import { t } from 'i18next';
import { IClaimHistory } from '../../models/helperFunctions/claimHistory';
import ClaimHistory from '../ClaimHistory';
import Button from '../Button';
import { ReactComponent as NotifySvg } from '../../../assets/NoNotification.svg';
import { ReactComponent as Block } from '../../../assets/Block.svg';
import ClaimTable from '../ClaimTable';
import Notifications from '../Notifications';
import Modal from '../Modal';
import ErrorMessage from '../ErrorMessge';
import React, { useEffect, useState } from 'react';
import { InputDescription } from '../AskManagerForm';
import { RolesTypes } from '@mimir/global-types';
import { toast } from 'react-toastify';
import AnswerToUser from '../AnswerToUser';
import { nanoid } from '@reduxjs/toolkit';

const InlineWrapper = styled.div`
  display: flex;
  flex-direction: row;
  column-gap: 4px;
  align-items: center;
`;
const ColumnWrapper = styled.div`
  display: flex;
  flex-direction: column;
  row-gap: inherit;
`;

const CardWrapper = styled(InlineWrapper)`
  background: ${colors.bg_secondary};
  height: 250px;
  width: auto;
  box-shadow: ${colors.shadow};
  border-radius: 10px;
  padding: 32px;
`;

const Avatar = styled.img`
  display: flex;
  width: 115px;
  height: 186px;
  object-fit: cover;
`;

const DescriptionWrapper = styled(ColumnWrapper)`
  margin-left: 24px;
  font-size: ${dimensions.base};
  row-gap: 8px;
`;

interface IDescriptionProps {
  bold?: boolean;
  titlee?: boolean;
  small?: boolean;
}

export const Description = styled.p<IDescriptionProps>`
  font-weight: ${({ bold, titlee }) => (bold ? (titlee ? 700 : 500) : 300)};
  font-size: ${({ titlee }) =>
    titlee ? `${dimensions.xl_2}` : `${dimensions.base}`};
  line-height: ${({ titlee }) =>
    titlee ? `${dimensions.xl_2}` : `${dimensions.xl}`};
  margin-bottom: ${({ titlee }) => (titlee ? dimensions.base : null)};
`;

export const Title = styled.h1`
  font-weight: 700;
  font-size: ${dimensions.xl_2};
  margin-bottom: ${dimensions.base};
  margin-top: ${dimensions.base_2};
`;

const ButtonsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: auto;
  margin-left: auto;
  max-width: 276px;
  width: 100%;
  row-gap: 8px;
`;

const RadioButton = styled.input`
  display: flex;
  margin-left: auto;
  margin-right: 414px;
  appearance: none;
  border: 1px solid ${colors.main_gray};
  border-radius: 50%;
  width: 24px;
  height: 24px;

  &:checked {
    background-color: ${colors.accent_color};
    outline-offset: -5px;
    outline: 4px solid ${colors.bg_secondary};
    border-color: ${colors.accent_color};
  }

  &:hover {
    border: 1px solid ${colors.description_gray};
  }
`;

const Form = styled.form`
  display: flex;
  flex-direction: column;
  row-gap: ${dimensions.base};
`;

const StyledButton = styled(Button)`
  :hover {
    border-color: ${colors.hover_color};
  }
`;

const UserCard = () => {
  const { id } = useParams();
  const {
    data: OnePerson,
    loading,
    error,
  } = useGetOnePersonQuery({
    variables: { id: id! },
  });
  const [setState] = useCreateStateMutation({
    refetchQueries: [GetOnePersonDocument],
  });
  const [changePersonRole] = useChangePersonRoleMutation({
    refetchQueries: [GetOnePersonDocument],
  });
  const [checkedButton, setCheckedButton] = useState<number>(0);
  const [showWarningBlock, setShowWarningBlock] = useState<boolean>(false);
  const [showWarningUnblock, setShowWarningUnblock] = useState<boolean>(false);
  const [showBlockInput, setShowBlockInput] = useState<boolean>(false);
  const [isNotificationModalOpen, setIsNotificationModalOpen] = useState(false);
  const [value, setValue] = useState<string>('');
  const messages = OnePerson?.getOnePerson.messages?.map((message) => {
    return {
      id: message?.id,
      type: 'message',
      created_at: message?.created_at,
      title: message!.title,
      message: message?.message,
    };
  });
  const states = OnePerson?.getOnePerson.states?.map((state) => {
    return {
      id: state?.id,
      type: 'block',
      created_at: state?.created_at,
      title: state?.state
        ? 'User have been blocked'
        : 'User have been unblocked',
      message: state?.description ? state?.description : '',
    };
  });

  const sortedNotifications = messages?.concat(states!);

  const handleChangeValue = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    setValue(e.target.value);
  };

  const handleChangePersonRoleClick = async () => {
    if (typeof id === 'undefined') return;
    const newType =
      OnePerson?.getOnePerson.type === RolesTypes.READER
        ? RolesTypes.MANAGER
        : RolesTypes.READER;
    try {
      await changePersonRole({
        variables: { person_id: parseInt(id), type: newType },
      });
    } catch (e) {
      toast.error((e as Error).message);
    }
  };

  const openNotificationModal = () => {
    setIsNotificationModalOpen(true);
  };

  const OpenModal = () => {
    if (state) {
      setShowWarningUnblock(true);
    } else setShowWarningBlock(true);
  };

  const blockSubmit = async () => {
    let sendValue = null;
    switch (checkedButton) {
      case 1:
        sendValue = 'Too many debts';
        break;
      case 2:
        sendValue = 'The user has been fired';
        break;
      case 3:
        sendValue = value;
        break;
    }

    await setState({
      variables: {
        person_id: Number(OnePerson?.getOnePerson.id),
        state: true,
        description: sendValue,
      },
    });
    setShowBlockInput(false);
  };

  const unblockSubmit = async () => {
    await setState({
      variables: {
        person_id: Number(OnePerson?.getOnePerson.id),
        state: false,
      },
    });
    setShowWarningUnblock(false);
  };

  const openInput = () => {
    setShowWarningBlock(false);
    setShowBlockInput(true);
  };

  useEffect(() => {
    if (error) toast.error(error.message);
  }, [error]);

  const READY_MADE_ANSWERS = [
    t('UserCard.NotificationAnswers.MissedDate'),
    t('UserCard.NotificationAnswers.AcceptedDonation'),
    t('UserCard.NotificationAnswers.Banned'),
  ];

  const state = OnePerson?.getOnePerson.states?.slice().pop()?.state;
  if (loading) return <h1>{t('Loading')}</h1>;

  return (
    <div style={{ marginBottom: '138px' }}>
      <BackButton />
      <CardWrapper data-testid="user-card">
        <Avatar src={OnePerson?.getOnePerson.avatar || mockData.avatar} />
        <DescriptionWrapper>
          <Description bold titlee>
            {OnePerson?.getOnePerson.username}
          </Description>
          <InlineWrapper>
            <Description bold>{t('UserCard.Position')}</Description>
            <Description>{OnePerson?.getOnePerson.position}</Description>
          </InlineWrapper>
          <InlineWrapper>
            <Description bold>E-mail:</Description>
            <Description>{OnePerson?.getOnePerson.email}</Description>
          </InlineWrapper>
          <ClaimHistory
            statuses={OnePerson?.getOnePerson.statuses as IClaimHistory[]}
          />
        </DescriptionWrapper>
        <ButtonsWrapper>
          <StyledButton
            data-testid="notification-btn"
            value={t('UserCard.CreateNotification')}
            onClick={openNotificationModal}
            svgComponent={<NotifySvg />}
            transparent
          ></StyledButton>
          {state ? (
            <Button
              value={t('UserCard.UnblockUser')}
              secondary
              warning
              onClick={OpenModal}
              svgComponent={<Block />}
            ></Button>
          ) : (
            <Button
              value={t('UserCard.BlockUser')}
              secondary
              warning
              transparent
              onClick={OpenModal}
              svgComponent={<Block />}
            ></Button>
          )}
          <StyledButton
            value={
              OnePerson?.getOnePerson.type === RolesTypes.READER
                ? t('UserCard.MakeManager')
                : t('UserCard.MakeReader')
            }
            transparent
            onClick={handleChangePersonRoleClick}
          />
        </ButtonsWrapper>
      </CardWrapper>
      <Title>{t('UserCard.ClaimList')}</Title>
      <Description>{t('UserCard.ClaimListDescription')}</Description>
      <ClaimTable
        statuses={OnePerson?.getOnePerson.statuses as IClaimHistory[]}
        name={OnePerson?.getOnePerson.username as string}
      />
      {!!sortedNotifications?.length && (
        <Notifications notifications={sortedNotifications}></Notifications>
      )}
      <Modal
        active={isNotificationModalOpen}
        setActive={setIsNotificationModalOpen}
      >
        <AnswerToUser
          close={() => setIsNotificationModalOpen(false)}
          answers={READY_MADE_ANSWERS}
          person_id={OnePerson?.getOnePerson.id}
          isSimpleNotification={true}
        />
      </Modal>
      <Modal active={showWarningUnblock} setActive={setShowWarningUnblock}>
        <ErrorMessage
          title={t('Block.Warning')}
          message={
            t('Block.SureUnblock') +
            OnePerson?.getOnePerson.username +
            t('Block.InLibrary')
          }
          titleCancel={t('Cancel')}
          setActive={setShowWarningUnblock}
          titleOption={t('Block.YUnblock')}
          onSubmitClick={unblockSubmit}
        />
      </Modal>
      <Modal active={showWarningBlock} setActive={setShowWarningBlock}>
        <ErrorMessage
          title={t('Block.Warning')}
          message={
            t('Block.SureBlock') +
            OnePerson?.getOnePerson.username +
            t('Block.InLibrary')
          }
          titleCancel={t('Cancel')}
          setActive={setShowWarningBlock}
          titleOption={t('Block.YBlock')}
          onSubmitClick={openInput}
        />
      </Modal>
      <Modal active={showBlockInput} setActive={setShowBlockInput}>
        <Form onSubmit={blockSubmit}>
          <Description bold titlee>
            {t('Block.BlockUser')}
          </Description>
          <Description bold>{t('Block.Reason')}</Description>

          <InlineWrapper>
            <Description>{t('Block.TooManyDebts')}</Description>
            <RadioButton
              type={'radio'}
              name={'description'}
              checked={checkedButton === 1}
              onChange={() => setCheckedButton(1)}
            />
          </InlineWrapper>
          <InlineWrapper>
            <Description>{t('Block.Fired')}</Description>
            <RadioButton
              type={'radio'}
              name={'description'}
              checked={checkedButton === 2}
              onChange={() => setCheckedButton(2)}
            />
          </InlineWrapper>
          <InlineWrapper>
            <Description>{t('Block.Other')}</Description>
            <RadioButton
              type={'radio'}
              name={'description'}
              checked={checkedButton === 3}
              onChange={() => setCheckedButton(3)}
            />
          </InlineWrapper>

          <Description bold>{t('Block.FullOther')}</Description>
          <InputDescription
            placeholder={'Enter your message'}
            disabled={checkedButton !== 3}
            value={value}
            onChange={handleChangeValue}
          />
          <InlineWrapper>
            <Button value={t('Block.BlockUser')} type={'submit'} />
            <Button value={t('Cancel')} transparent />
          </InlineWrapper>
        </Form>
      </Modal>
    </div>
  );
};

export default UserCard;
