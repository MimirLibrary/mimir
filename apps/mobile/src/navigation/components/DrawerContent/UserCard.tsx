import React from 'react';
import { AppText } from '../../../components/AppText';
import { TextVariant } from '../../../config/textVariants';
import { StyleSheet, View } from 'react-native';
import { Avatar } from '../../../components/Avatar';
import { Fonts } from '../../../assets/fonts';
import { ButtonIcon } from '../../../components/ButtonIcon';
import { CloseIcon } from '../../../components/icons/CloseIcon';

const DUMMY_DATA =
  'https://avatars.mds.yandex.net/i?id=69223b715b8bde25b0473fef8d40cb6c-5488638-images-thumbs&n=13&exp=1';

export const UserCard = () => {
  return (
    <View style={styles.userCardContainer}>
      <View
        style={{
          flexDirection: 'row',
          width: '100%',
          justifyContent: 'space-between',
          alignItems: 'center',
        }}
      >
        <AppText variant={TextVariant.H1} text={'MIMIR'} />
        <ButtonIcon icon={<CloseIcon />} onPress={() => {}} />
      </View>
      <View style={styles.avatarContainer}>
        <Avatar avatar={DUMMY_DATA} containerStyle={styles.avatar} />
        <AppText
          variant={TextVariant.H2}
          text={'Ivan Ivanov'}
          style={styles.text}
        />
      </View>
      <AppText
        variant={TextVariant.CAPTION_R}
        text={`ivan.ivanov@itechart\n-group.com`}
        style={[styles.description, styles.text]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  userCardContainer: {
    paddingTop: 16,
    paddingHorizontal: 16,
    paddingBottom: 56,
    alignItems: 'center',
  },
  avatarContainer: { marginTop: 38, alignItems: 'center' },
  text: { textAlign: 'center' },
  avatar: { width: 80, height: 80, marginBottom: 16 },
  description: {
    marginTop: 8,
  },
});
