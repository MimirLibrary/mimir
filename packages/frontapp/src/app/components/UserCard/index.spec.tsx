import { render } from '../../../helpers/customRender';
import { screen } from '@testing-library/react';
import UserCard from './index';
import { GetOnePersonDocument } from '@mimir/apollo-client';
import { MockedProvider } from '@apollo/client/testing';
import { nextTick } from '../../../helpers/tests/nextTick';

const personMock = {
  request: {
    query: GetOnePersonDocument,
  },
  result: {
    data: {
      getOnePerson: {
        avatar: '',
        email: 'pushking@gmail.com',
        id: '4',
        messages: [],
        position: 'JS Developer',
        states: [],
        statuses: [],
        type: 'Reader',
        username: 'Pete',
      },
    },
  },
};

jest.mock('i18next', () => ({
  t: (i: any) => i,
}));

describe('UserCard Component', () => {
  beforeEach(() => {
    render(
      <MockedProvider mocks={[personMock]} addTypename={false}>
        <UserCard />
      </MockedProvider>
    );
  });

  it('should render loading', async () => {
    expect(screen.getByText('Loading')).toBeInTheDocument();
  });

  it('should render result', async () => {
    await nextTick();

    expect(screen.getByTestId('user-card')).toBeInTheDocument();
    expect(screen.getByText('Pete')).toBeInTheDocument();
    expect(screen.getByText('JS Developer')).toBeInTheDocument();
    expect(screen.getByText('pushking@gmail.com')).toBeInTheDocument();
    expect(screen.getByTestId('notification-btn')).toBeInTheDocument();
  });

  it('should match snapshot', async () => {
    await nextTick();

    expect(screen.getByTestId('user-card')).toMatchSnapshot();
  });
});
