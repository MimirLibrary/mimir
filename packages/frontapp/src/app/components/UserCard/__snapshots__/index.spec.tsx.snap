// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard Component should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-column-gap: 4px;
  column-gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  height: 250px;
  width: auto;
  box-shadow: 0px 10px 70px rgba(26, 30, 214, 0.08);
  border-radius: 10px;
  padding: 32px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 115px;
  height: 186px;
  object-fit: cover;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: inherit;
  margin-left: 24px;
  font-size: 1rem;
  row-gap: 8px;
}

.emotion-3 {
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 1.5rem;
  margin-bottom: 1rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-column-gap: 4px;
  column-gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.25rem;
}

.emotion-6 {
  font-weight: 300;
  font-size: 1rem;
  line-height: 1.25rem;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: inherit;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-column-gap: 4px;
  column-gap: 4px;
}

.emotion-12 {
  font-weight: 500;
  font-size: inherit;
  line-height: 1.25rem;
}

.emotion-13 {
  font-weight: 300;
  font-size: inherit;
  line-height: 1.25rem;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: auto;
  margin-left: auto;
  max-width: 276px;
  width: 100%;
  row-gap: 8px;
}

.emotion-21 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border-radius: 1.875rem;
  height: 3.125rem;
  width: 100%;
  border: 2px solid #1A1ED6;
}

.emotion-21:hover {
  background-color: #FFFFFF;
  border: 2px solid #1A1DB1;
}

.emotion-21:hover svg {
  fill: #1A1DB1;
}

.emotion-21:hover span {
  color: #1A1DB1;
}

.emotion-21:active {
  background-color: #FFFFFF;
  border: 2px solid #1A1DB1;
}

.emotion-21:active svg {
  fill: #1A1DB1;
}

.emotion-21:active span {
  color: #1A1DB1;
}

.emotion-21:disabled {
  background: #BDBDBD;
  cursor: default;
}

.emotion-21 svg {
  fill: #1A1DB1;
  height: auto;
  max-width: 1.5rem;
}

.emotion-21 span {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.25rem;
  color: #1A1DB1;
  margin-left: 0.625rem;
}

.emotion-21:hover {
  border-color: #1A1DB1;
}

.emotion-22 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border-radius: 1.875rem;
  height: 3.125rem;
  width: 100%;
  border: 2px solid #BDBDBD;
}

.emotion-22:hover {
  background-color: #FFFFFF;
  border: 2px solid #FF372A;
}

.emotion-22:hover svg {
  fill: #FF372A;
}

.emotion-22:hover span {
  color: #FF372A;
}

.emotion-22:active {
  background-color: #FF372A;
  border: 2px solid #FF372A;
}

.emotion-22:active svg {
  fill: #FFFFFF;
}

.emotion-22:active span {
  color: #FFFFFF;
}

.emotion-22:disabled {
  background: #BDBDBD;
  cursor: default;
}

.emotion-22 svg {
  fill: #BDBDBD;
  height: auto;
  max-width: 1.5rem;
}

.emotion-22 span {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.25rem;
  color: #BDBDBD;
  margin-left: 0.625rem;
}

.emotion-23 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border-radius: 1.875rem;
  height: 3.125rem;
  width: 100%;
  border: 2px solid #1A1ED6;
}

.emotion-23:hover {
  background-color: #FFFFFF;
  border: 2px solid #1A1DB1;
}

.emotion-23:hover svg {
  fill: #1A1DB1;
}

.emotion-23:hover span {
  color: #1A1DB1;
}

.emotion-23:active {
  background-color: #FFFFFF;
  border: 2px solid #1A1DB1;
}

.emotion-23:active svg {
  fill: #1A1DB1;
}

.emotion-23:active span {
  color: #1A1DB1;
}

.emotion-23:disabled {
  background: #BDBDBD;
  cursor: default;
}

.emotion-23 svg {
  fill: #1A1DB1;
  height: auto;
  max-width: 1.5rem;
}

.emotion-23 span {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.25rem;
  color: #1A1DB1;
}

.emotion-23:hover {
  border-color: #1A1DB1;
}

<div
  class="emotion-0"
  data-testid="user-card"
>
  <img
    class="emotion-1"
    src="avatar.jpg"
  />
  <div
    class="emotion-2"
  >
    <p
      class="emotion-3"
    >
      Pete
    </p>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
      >
        UserCard.Position
      </p>
      <p
        class="emotion-6"
      >
        JS Developer
      </p>
    </div>
    <div
      class="emotion-4"
    >
      <p
        class="emotion-5"
      >
        E-mail:
      </p>
      <p
        class="emotion-6"
      >
        pushking@gmail.com
      </p>
    </div>
    <div
      class="emotion-10"
    >
      <div
        class="emotion-11"
      >
        <p
          class="emotion-12"
        >
          Readers.SingleUser.ClaimHistory
        </p>
        <p
          class="emotion-13"
        >
          -
        </p>
      </div>
      <div
        class="emotion-11"
      >
        <p
          class="emotion-12"
        >
          Readers.SingleUser.ClaimNow
        </p>
        <p
          class="emotion-13"
        >
          -
        </p>
      </div>
      <div
        class="emotion-11"
      >
        <p
          class="emotion-12"
        >
          Statuses.Overdue: 
        </p>
        <p
          class="emotion-13"
        >
          -
        </p>
      </div>
    </div>
  </div>
  <div
    class="emotion-20"
  >
    <button
      class="emotion-21"
      data-testid="notification-btn"
      type="button"
      value="UserCard.CreateNotification"
    >
      <svg>
        NoNotification.svg
      </svg>
      <span>
        UserCard.CreateNotification
      </span>
    </button>
    <button
      class="emotion-22"
      type="button"
      value="UserCard.BlockUser"
    >
      <svg>
        Block.svg
      </svg>
      <span>
        UserCard.BlockUser
      </span>
    </button>
    <button
      class="emotion-23"
      type="button"
      value="UserCard.MakeManager"
    >
      <span>
        UserCard.MakeManager
      </span>
    </button>
  </div>
</div>
`;
